k3k-kubelet/provider/provider.go

```diff
@@ -193,75 +221,223 @@ func (p *Provider) PortForward(ctx context.Context, namespace, pod string, port
 
 // CreatePod takes a Kubernetes Pod and deploys it within the provider.
 func (p *Provider) CreatePod(ctx context.Context, pod *corev1.Pod) error {
-	translated := p.translateTo(pod)
+	tPod := pod.DeepCopy()
+	p.Translater.TranslateTo(tPod)
+
 	// these values shouldn't be set on create
-	translated.UID = ""
-	translated.ResourceVersion = ""
-	// can't handle volumes yet, just omit them for now
-	translated.Spec.Volumes = nil
-	containers := []corev1.Container{}
-	for _, container := range translated.Spec.Containers {
-		container.VolumeMounts = nil
-		containers = append(containers, container)
-	}
-	translated.Spec.Containers = containers
-	translated.Spec.NodeName = ""
-	p.logger.Infof("Creating pod %s/%s for pod %s/%s", translated.Namespace, translated.Name, pod.Namespace, pod.Name)
-	_, err := p.CoreClient.Pods(p.ClusterNamespace).Create(ctx, translated, metav1.CreateOptions{})
-	return err
+	tPod.UID = ""
+	tPod.ResourceVersion = ""
+
+	// the node was scheduled on the virtual kubelet, but leaving it this way will make it pending indefinitely
+	tPod.Spec.NodeName = ""
+
+	// volumes will often refer to resources in the virtual cluster, but instead need to refer to the sync'd
+	// host cluster version
+	err := p.transformVolumes(ctx, pod.Namespace, tPod.Spec.Volumes)
+	if err != nil {
+		return fmt.Errorf("unable to sync volumes for pod %s/%s: %w", pod.Namespace, pod.Name, err)
+	}
+	p.logger.Infof("Creating pod %s/%s for pod %s/%s", tPod.Namespace, tPod.Name, pod.Namespace, pod.Name)
```

Comment:
just to be consistent with the rest of the code I think we can just use 

```suggestion
	p.logger.Infow("Creating pod", "Translated Namespace", tPod.Namespace, "Translated Name", tPod.Name, "Namespace", pod.Namespace, "Name", pod.Name)
```